contracts/SportMarkets/Parlay/ParlayVerifier.sol

```diff
@@ -552,115 +542,40 @@ contract ParlayVerifier {
-    {
-        uint[] memory buyQuoteAmountPerMarket = new uint[](params.sportMarkets.length);
-        buyAmountPerMarket = params.buyQuoteAmounts;
-        finalQuotes = new uint[](params.sportMarkets.length);
-        uint feesIncluded;
-        for (uint i = 0; i < params.sportMarkets.length; i++) {
-            totalBuyAmount += params.buyQuoteAmounts[i];
-            // buyQuote always calculated with added SportsAMM fees
-            buyQuoteAmountPerMarket[i] = (params.defaultONE *
-                params.sportsAmm.buyFromAmmQuote(
-                    params.sportMarkets[i],
-                    obtainSportsAMMPosition(params.positions[i]),
-                    params.buyQuoteAmounts[i]
-                ));
-            if (buyQuoteAmountPerMarket[i] == 0) {
-                totalQuote = 0;
-                totalBuyAmount = 0;
-            }
-        }
-        for (uint i = 0; i < params.sportMarkets.length; i++) {
-            finalQuotes[i] = ((buyQuoteAmountPerMarket[i] * ONE * ONE) / params.buyQuoteAmounts[i]) / ONE;
-            // where the SGP fee is actually applied
-            if (params.sgpFees[i] > 0) {
-                finalQuotes[i] = ((finalQuotes[i] * ONE * ONE) / params.sgpFees[i]) / ONE;
-                feesIncluded += params.sgpFees[i] > ONE ? params.sgpFees[i] - ONE : (ONE - params.sgpFees[i]);
-            }
-            totalQuote = (i == 0) ? finalQuotes[i] : (totalQuote * finalQuotes[i]) / ONE;
-        }
-        if (totalQuote > 0) {
-            if (totalQuote < IParlayMarketsAMM(params.sportsAmm.parlayAMM()).maxSupportedOdds()) {
-                totalQuote = IParlayMarketsAMM(params.sportsAmm.parlayAMM()).maxSupportedOdds();
-            }
-            uint expectedPayout = ((params.sUSDAfterFees * ONE * ONE) / totalQuote) / ONE;
-            skewImpact = expectedPayout > totalBuyAmount
-                ? (((ONE * expectedPayout) - (ONE * totalBuyAmount)) / (totalBuyAmount))
-                : (((ONE * totalBuyAmount) - (ONE * expectedPayout)) / (totalBuyAmount));
-            buyAmountPerMarket = _applySkewImpactBatch(buyAmountPerMarket, skewImpact, (expectedPayout > totalBuyAmount));
-            totalBuyAmount = _applySkewImpact(totalBuyAmount, skewImpact, (expectedPayout > totalBuyAmount));
-            _calculateRisk(params.sportMarkets, (totalBuyAmount - params.sUSDAfterFees), params.sportsAmm.parlayAMM());
-            if (feesIncluded > 0) {
-                if (skewImpact > feesIncluded) {
-                    skewImpact = skewImpact - feesIncluded;
-                } else {
-                    skewImpact = feesIncluded - skewImpact;
+            for (uint i = 0; i < numOfMarkets; i++) {
+                //consider if this works well for Arbitrum at 6 decimals
+                if (finalQuotes[i] > 0) {
+                    amountsToBuy[i] = (ONE * params.totalSUSDToPay) / finalQuotes[i];
                 }
             }
-        } else {
-            totalBuyAmount = 0;
         }
     }
 
```

Comment:
Functions that can be deleted in ParlayVerifier:
`_applySkewImpact()`
`_applySkewImpactBatch()`
`obtainSportsAMMPosition()`

Reply:
obtainSportsAMMPosition is used in ParlayVault though